src/main/java/duke/CommandParserAndLogic.java

```diff
@@ -0,0 +1,140 @@
+package duke;
+
+import duke.Tasks.DeadLineTask;
+import duke.Tasks.EventTask;
+import duke.Tasks.TodoTask;
+import duke.TextStoreAndPrint.TextPrinter;
+
+import java.time.format.DateTimeParseException;
+
+public class CommandParserAndLogic {
+
+    TaskList taskList;
+    String[] current;
+
+    public CommandParserAndLogic(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Splits the string and uses commandCheck method for further logic
+     *
+     * @param input a string input
+     */
+    public void handleInput(String input) {
+        current = input.split(" ", 2);
+        commandCheck(current[0]);
+    }
+
+    /**
+     * Checks the string provided if it matches any known commands
+     * Follows up by calling the relevant logic
+     *
+     * @param string input
+     * @return true if it is an know command, else false
+     */
+    private boolean commandCheck(String string) {
```

Comment:
Maybe can put checkCommand instead, like standardizing the verb first or vice versa

Reply:
Ill keep naming conventions in mind, thanks.